pkg/controller/endpointslice/utils.go

```diff
@@ -169,6 +170,16 @@ func newEndpointSlice(service *corev1.Service, endpointMeta *endpointMeta) *disc
 	}
 }
 
+// getEndpointSlicePrefix returns a suitable prefix for an EndpointSlice name.
+func getEndpointSlicePrefix(serviceName string) string {
+	// use the dash (if the name isn't too long) to make the pod name a bit prettier
```

Comment:
so this is display-only? nothing tries to parse the service name out of the endpointslice name? glancing at the KEP, it seemed like a client could assume the `<service>.<suffix>` format and parse things from it

Reply:
Correct, that's what the label is for. The KEP needs to be updated here, notably to include the `-` naming scheme as opposed to the invalid `.` scheme it originally included. This new logic matches a pattern already commonly used by pods, so hopefully it makes sense.

Reply:
ok, probably worth clarifying in the KEP that the name isn't appropriate for reverse-engineering the service and the label is authoritative